src/codegen/function_body.rs

```diff
@@ -0,0 +1,49 @@
+use std::fmt;
+
+use super::general::tabs;
+
+macro_rules! write_to_vec(
+    ($dst:expr, $($arg:tt)*) => (
+        $dst.push(fmt::format(format_args!($($arg)*)))
+    )
+);
+
+#[derive(Default, Debug)]
+pub struct Builder {
+    glib_name: String,
+    from_glib_prefix: String,
+    from_glib_suffix: String,
+    parameters: Vec<String>,
+}
+
+impl Builder {
+    pub fn new() -> Builder {
+        Default::default()
+    }
+    pub fn glib_name<'a>(&'a mut self, name: &str) -> &'a mut Builder {
```

Comment:
Did lifetime elision not handle this correctly? Reading the [TRPL](https://doc.rust-lang.org/book/lifetimes.html#lifetime-elision), it seems the return value would get the same lifetime as `self`.